mixer/adapter/kubernetesenv/kubernetesenv.go

```diff
@@ -189,7 +226,16 @@ func (h *handler) Close() error {
 
 func (h *handler) findPod(uid string) (*v1.Pod, bool) {
 	podKey := keyFromUID(uid)
-	pod, found := h.k8sCache.Pod(podKey)
+	var found bool
+	var pod *v1.Pod
+
+	for _, controller := range h.k8sCache {
```

Comment:
Are we sure this is the case?  I don't actually know the real answer, but I *think* the pod name is distinct from the pod UID.

For deployment-generated pods, it feels like the pod name is `<replicaset name> + '-' + <random chars>`. Looking at a running cluster, I see pod `prometheus-6c9fdd4bb-p29r4` comes from `ReplicaSet/prometheus-6c9fdd4bb`.

If I hand-crafted a ReplicaSet to be named 'test', I'm betting the pods would be `test-<5 chars>`, which isn't a UUID.

If i wrote *just* a PodSpec (instead of a deployment or replicaset), I'm betting the name would be even less unique (no random chars).

I'm not sure how much any of this matters in reality.  I guess I was just wondering if we need to start putting cluster names in the `uid` here (and then selecting the appropriate cache). This probably is related to the follow-on work required to populate cluster name attributes, etc.

Reply:
Yeah, if you created your own ReplicaSet then you would be limited to the uniqueness of the last five values. These seem to be mini-UUIDs (although I don't know exactly how k8s generates). These five digits do consist of numbers and letters so that would make (36^5) uniqueness. 
I'm not familiar w/ k8s PodSpec, but it does seem like that may create same named pods between multi-clusters. Could be an issue from a mixer perspective. Something we'll need to keep in mind.